src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -504,177 +506,179 @@ public static partial class TensorPrimitives
     }
     public static partial class TensorSpan
```

Comment:
I think we should talk about the `Tensor` vs `TensorSpan` class split and how it impacts discoverability and usability of the APIs -- CC. @stephentoub 

Right now we basically have (for example):
```csharp
static class Tensor
{
    public static Tensor<T> Abs<T>(Tensor<T> input);
    public static Tensor<T> AbsInPlace<T>(Tensor<T> input);
}

static class TensorSpan
{
    public static TensorSpan<T> Abs<T>(in ReadOnlyTensorSpan<T> input);
    public static TensorSpan<T> AbsInPlace<T>(in TensorSpan<T> input);
}
```

The former APIs typically refer to the latter APIs (`Tensor<T>` defers to `TensorSpan<T>`) but the latter APIs sometimes need to allocate a new buffer and are returning that as a span, which is somewhat problematic. It also lacks the ability for users to support their own buffers.


I think rather we should set this up as:
```csharp
static class Tensor
{
    public static Tensor<T> Abs<T>(in ReadOnlyTensorSpan<T> input);
    public static void Abs<T>(in ReadOnlyTensorSpan<T> input, in TensorSpan<T> destination);
    public static void AbsInPlace<T>(in TensorSpan<T> input);
}
```

This minimizes the total number of APIs we need to expose, ensures that users can take ownership of the allocated buffer in case we need to allocate one, and that they can provide their own buffer in case they want to customize how that is allocated.

It's worth noting that the latter two APIs are returning `void` just like `TensorPrimitives` returns `void` because that's the default expectation for most similar APIs. Returning the `destination` buffer (which is `input` in the case of `AbsInPlace`) is a matter of convenience and allows for easier chaining and more fluent like API usage. -- I don't have a particular preference here and personally lean towards the more fluent like approach, but I expect it will entail discussion in API review either way.